src/Npgsql/Schema/DbColumnSchemaGenerator.cs

```diff
@@ -35,7 +35,12 @@ internal DbColumnSchemaGenerator(NpgsqlConnection connection, RowDescriptionMess
      {(pgVersion.IsGreaterOrEqual(10) ? "attidentity != ''" : "FALSE")} AS isidentity,
      CASE WHEN typ.typtype = 'd' THEN typ.typtypmod ELSE atttypmod END AS typmod,
      CASE WHEN atthasdef THEN (SELECT pg_get_expr(adbin, cls.oid) FROM pg_attrdef WHERE adrelid = cls.oid AND adnum = attr.attnum) ELSE NULL END AS default,
-     CASE WHEN col.is_updatable = 'YES' THEN true ELSE false END AS is_updatable,
+     CASE WHEN ((cls.relkind = ANY (ARRAY['r'::""char"", 'p'::""char""])) 
```

Comment:
Not super happy about swapping a nice, high-level "is_updatable" check for a lower-level thing like this, where it's our responsibility to make sure we get everything just right... Getting the schema is generally not a very high-perf thing (after all it does a full database roundtrip), so I'm not sure reducing the join here justifies it.

What's your opinions @vonzshik, @NinoFloris, @Brar?

Reply:
It is the same logic as the Postgres view that creates the meta data tables, so it is not guess work, it simply needs to be kept in sync. Additionally, the views are horribly slow in comparison for operations that require the meta data from a statement.

Reply:
OK, maybe it's OK then... It's true that this is very unlikely to change, and we can handle any change based on user reports if/when it does happen...

Reply:
Perhaps the real question is why isn't all of the meta data based on a materialized view with added indexes so that they perform well.  This would be a much better solution because it would keep the logic centralized.

Reply:
@DruAtNexee you're referring to the PostgreSQL information_schema views, right? If so, that definitely sounds like something that's worth bringing up in the PG hackers mailing list.

Reply:
Yes, I am referring to the information_schema views, querying those table directly perform horribly and I wonder if utilizing materialized views coupled with adding properly indexes to the resulting views would fix the performance problems.  This would keep the meta data (information_schema) tables abstract and centralize the logic.

Reply:
> Not super happy about swapping a nice, high-level "is_updatable" check for a lower-level thing like this, where it's our responsibility to make sure we get everything just right... Getting the schema is generally not a very high-perf thing (after all it does a full database roundtrip), so I'm not sure reducing the join here justifies it.

While it's definitely not an amazing thing, it does improve query time quite a bit (which should help with startup time for FaaS) and we do test against the latest 5 PG releases, so we should be able to catch most of the errors with our CI.